### PR TITLE
Support building and pushing multi-platform registry image

### PR DESCRIPTION
**Pull request by pkoutsovasilis**
This PR builds `package-registry` container image as a multi-platform one, supporting linux/amd64 and linux/arm64, by utilising `docker buildx`. Having such an image of package-registry will help us achieve native execution of `elastic-package stack up` on arm64-based machines.

PS: I am not that experienced in `buildkite` to make sure that `buildx` is always installed and available on the runner 🙂

### CODE REVIEW

**Review by mrodm — COMMENTED** [review]
Thanks @pkoutsovasilis ! 

It'd be great to have this support in the docker images generated :muscle: Added some suggestions and doubts .

Pinging @jsoriano to take another look on this change, in case there is something missing from my side.

File: .buildkite/scripts/publish-distribution.sh [hunk 1]
```diff
@@ -14,16 +14,13 @@ else
...
+docker buildx create --use
 
 # do not push if DRY_RUN is true
 if [[ ${DRY_RUN:-true} == "true" ]]; then
-    echo "Docker push command will be: retry 3 docker push ${DOCKER_IMG_TARGET}"
+    docker buildx imagetools create --dry-run -t "${DOCKER_IMG_TARGET}" "${DOCKER_IMG_SOURCE}"
 else
-    echo "Docker push:"
-    retry 3 docker push "${DOCKER_IMG_TARGET}"
+    docker buildx imagetools create -t "${DOCKER_IMG_TARGET}" "${DOCKER_IMG_SOURCE}"
```
[the remark this answers is not in the document]
Comment (pkoutsovasilis) [reply .buildkite/scripts/publish-distribution.sh:24]: we can surely add it back although I am not 100% sure why we would want to force retry; if I am not mistaken docker has an internal retry mechanism depending on the push error

File: .buildkite/scripts/publish.sh [hunk 2]
```diff
@@ -5,18 +5,18 @@ set -euo pipefail
 
 
 pushDockerImage() {
-    docker build \
+    docker buildx create --use
+    docker buildx build --push \
```
[the remark this answers is not in the document]
Comment (pkoutsovasilis) [reply .buildkite/scripts/publish.sh:9]: same as above

File: .buildkite/scripts/publish-distribution.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (pkoutsovasilis) [reply .buildkite/scripts/publish-distribution.sh:24]: it shouldn't cause any issue, with the docker buildx the same images should be produced but they will have multiple manifests with layers each targeting a different platform e.g.

```console
docker manifest inspect docker.elastic.co/observability-ci/package-registry:PR-1162
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.oci.image.index.v1+json",
   "manifests": [
      {
         "mediaType": "application/vnd.oci.image.manifest.v1+json",
         "size": 1242,
         "digest": "sha256:08e713258d5c33a47e54af0dc45d843bf1850c3cdcca4bc6123d1f586a1ff75e",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.oci.image.manifest.v1+json",
         "size": 1242,
         "digest": "sha256:566e1f33ec8b60e62c6ed3da5c00de16992934c2b79951d5d689fc99d8e48d32",
         "platform": {
            "architecture": "arm64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.oci.image.manifest.v1+json",
         "size": 566,
         "digest": "sha256:d88cc6b4286cb1e452f50da10b02bac3819d867086794015afca9fa49f320009",
         "platform": {
            "architecture": "unknown",
            "os": "unknown"
         }
      },
      {
         "mediaType": "application/vnd.oci.image.manifest.v1+json",
         "size": 566,
         "digest": "sha256:c3de05c83739b39d7f563fc2b76048d1635744bb451462e9d899cdb8b45b5e5e",
         "platform": {
            "architecture": "unknown",
            "os": "unknown"
         }
      }
   ]
}
```

File: .buildkite/scripts/publish.sh [hunk 3]
```diff
@@ -5,18 +5,18 @@ set -euo pipefail
...
         --label GO_VERSION="${SETUP_GOLANG_VERSION}" \
         --label TIMESTAMP="$(date +%Y-%m-%d_%H:%M)" \
         .
-    retry 3 docker push "${DOCKER_IMG_TAG}"
-    echo "Docker image pushed: ${DOCKER_IMG_TAG}"
-    docker tag "${DOCKER_IMG_TAG}" "${DOCKER_IMG_TAG_BRANCH}"
-    retry 3 docker push "${DOCKER_IMG_TAG_BRANCH}"
-    echo "Docker image pushed: ${DOCKER_IMG_TAG_BRANCH}"
+
+    echo "Docker images pushed: ${DOCKER_IMG_TAG} ${DOCKER_IMG_TAG_BRANCH}"
```
[the remark this answers is not in the document]
Comment (pkoutsovasilis) [reply .buildkite/scripts/publish.sh:19]: exactly the host will fetch the appropriate image based on manifest, shown above, and utilise this as the base

**Review by jsoriano — COMMENTED** [review]
LGTM, please add a changelog entry to CHANGELOG.md mentioning the new supported architectures.

I would sligthly prefer to keep the explicit retries, just in case.

I would also like to consider at some point the use of goreleaser, it also seems to support multi-platform docker images https://goreleaser.com/cookbooks/multi-platform-docker-images/
But to migrate to goreleaser we would also need to review how we generate the changelogs, so let's keep this for a separate change.

File: .buildkite/scripts/publish.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (pkoutsovasilis) [reply .buildkite/scripts/publish.sh]: added the `retry` 🙂

File: .buildkite/scripts/publish-distribution.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (pkoutsovasilis) [reply .buildkite/scripts/publish-distribution.sh]: `retry` added 🙂

[the remark this answers is not in the document]
Comment (pkoutsovasilis) [reply .buildkite/scripts/publish-distribution.sh]: @mrodm I did a manual testing `docker buildx imagetools create -t $DEST_IMG $SRC_IMG works for both single and multi-plarform images so I don't expect any issues here

File: CHANGELOG.md [hunk 4]
```diff
@@ -14,6 +14,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
+* Add support for multi-platform container images. [#1162](https://github.com/elastic/package-registry/pull/1162)
```
Comment (jsoriano) [inline CHANGELOG.md:17]: :+1:

**Review by jsoriano — APPROVED** [review]
Thanks!

**Review by mrodm — APPROVED** [review]
Thanks!

**Pull request merged**